File: algorithms/warmup/mini-max-sum/mini-max-sum.hs

numbers :: String -> [Int]
numbers nums = map (read::String->Int) (words nums)

min_i_max_aux count minimum maximum [] = return ((count - maximum), (count - minimum))
min_i_max_aux count minimum maximum (value:values) =
  min_i_max_aux (count + value) (min minimum value) (max maximum value) values

min_i_max (n:ns) = min_i_max_aux n n n ns

main :: IO()
main = do
  inputdata <- getContents
  (min, max) <- min_i_max (numbers inputdata)
  putStr $ show min ++ " " ++ show max

File: algorithms/warmup/mini-max-sum/mini-max-sum.rb
#!/bin/ruby

def min_i_max(numbers)
  maximum = minimum = numbers[0].to_i
  total_sum = numbers[0].to_i
  for n in numbers[1..-1]
    n_i = n.to_i
    total_sum += n_i
    maximum = [maximum, n_i].max
    minimum = [minimum, n_i].min
  end
  return (total_sum - maximum), (total_sum - minimum)
end

numbers   = gets.split(' ')
min_i_max = min_i_max(numbers)

print(min_i_max[0], ' ', min_i_max[1])
